# 📊 Cálculos e Análises

Bem-vindo à seção de **Cálculos** da dashboard!  
Aqui você encontrará todas as métricas, fórmulas e análises essenciais para interpretar os dados de forma clara e eficiente.

---

> **Explore os resultados, compare indicadores e tome decisões baseadas em dados!**

In [164]:
import pandas as pd

In [165]:
df = pd.read_csv("data/banco_agro_projeto_limpo.csv")

In [166]:
df

Produto Categoria  Estoque  Vendidos Data_de_vendas  \
0               Café    Cereal      466       277     2024-01-01   
1               Soja      Grão      191        37     2024-01-02   
2     Cana-de-açúcar      Grão      241       241     2024-01-03   
3              Arroz    Bebida      222       283     2024-01-04   
4              Arroz    Bebida      370       432     2024-01-05   
...              ...       ...      ...       ...            ...   
1995           Arroz    Cereal      493       336     2029-06-18   
1996  Cana-de-açúcar     Fibra      457       385     2029-06-19   
1997         Algodão     Fibra      316       276     2029-06-20   
1998           Trigo      Grão       65       319     2029-06-21   
1999           Milho     Fibra      982       403     2029-06-22   

          Vendedor          cidade   latitude  longitude  Preco_kg  
0     Desconhecido       São Paulo -23.550520 -46.633308     38.57  
1            Pedro  Rio de Janeiro -22.906847 -43.172896      2.33  
2         Fernanda        Brasília -15.794229 -47.882166      1.28  
3           Carlos  Belo Horizonte -19.924502 -43.935237      1.56  
4         Patrícia       Fortaleza  -3.717220 -38.543400      1.56  
...            ...             ...        ...        ...       ...  
1995      Patrícia       São Paulo -23.550520 -46.633308      1.56  
1996           Ana    Porto Alegre -30.034647 -51.217658      1.28  
1997         Maria          Recife  -8.047560 -34.877000     17.68  
1998          João        Salvador -12.971399 -38.501305      1.42  
1999       Ricardo          Recife  -8.047560 -34.877000      1.09  

[2000 rows x 10 columns]

In [167]:
df.describe()

Estoque     Vendidos     latitude    longitude     Preco_kg
count  2000.000000  2000.000000  2000.000000  2000.000000  2000.000000
mean    487.692000   259.006000   -16.501316   -45.410686     9.057880
std     287.213426   173.592126     8.915990     6.957261    12.940259
min       0.000000     1.000000   -30.034647   -60.025000     1.090000
25%     243.000000   119.000000   -23.550520   -49.267137     1.280000
50%     479.500000   244.000000   -19.924502   -46.633308     1.560000
75%     736.000000   377.000000    -8.047560   -38.543400    17.680000
max     999.000000   998.000000    -3.101940   -34.877000    38.570000

In [168]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Produto         2000 non-null   object 
 1   Categoria       2000 non-null   object 
 2   Estoque         2000 non-null   int64  
 3   Vendidos        2000 non-null   int64  
 4   Data_de_vendas  2000 non-null   object 
 5   Vendedor        2000 non-null   object 
 6   cidade          2000 non-null   object 
 7   latitude        2000 non-null   float64
 8   longitude       2000 non-null   float64
 9   Preco_kg        2000 non-null   float64
dtypes: float64(3), int64(2), object(5)
memory usage: 156.4+ KB


## Informações básicas

In [169]:
df['Vendidos'].sum() #total de produtos vendidos  

np.int64(518012)

In [170]:
df['Estoque'].sum() #total em estoque não vendido


np.int64(975384)

In [171]:
df['Preco_kg'].unique()

array([38.57,  2.33,  1.28,  1.56, 17.68,  1.42,  1.09])

In [172]:
df['Preco_kg'].mean() # preço médio por kg
# como temos dois valores com 38 e 17 eles aumentam e muito o valor dos nossos produtos, mas entretanto observamos
# que temois mais produtos baratos que caros 

np.float64(9.05788)

In [173]:
df.groupby('Produto')['Vendidos'].sum().reset_index().sort_values(by='Vendidos', ascending=False) # total vendido por produto  
# Percebemos que o algodão vende muito mesmo ele sendo extremamente barato

Produto  Vendidos
0         Algodão     84813
6           Trigo     82231
1           Arroz     73925
5            Soja     71977
3  Cana-de-açúcar     70430
4           Milho     67707
2            Café     66929

In [174]:
df.groupby('Produto')['Estoque'].sum().reset_index().sort_values(by='Estoque', ascending=False) # total de estoque por produto  
# Notamos que o trigo é oque mias está em estoque, mesmo ele vendendo muito, é possível repensar não só com o trigo mas com todos os outros produtos sobre se devemos produzir tanto

Produto  Estoque
6           Trigo   152750
0         Algodão   151739
3  Cana-de-açúcar   139702
5            Soja   138778
1           Arroz   136524
2            Café   128606
4           Milho   127285

## Faturamento sobre cada item da tabela

In [175]:
df['Produto'].unique()

array(['Café', 'Soja', 'Cana-de-açúcar', 'Arroz', 'Algodão', 'Trigo',
       'Milho'], dtype=object)

In [176]:

df['Faturamento'] = df['Vendidos'] * df['Preco_kg']
faturamento_total = df.groupby('Produto')['Faturamento'].sum().reset_index().round().sort_values(by="Faturamento", ascending=False)
faturamento_total
# Mesmo o algodão sendo mais vendido o café nos rende mais, mesmo sendo o que menos vende entre todos os produtos# Percebemos tambêm que o milho vende pouco e faturou pouco, logo é nescessário discutir sobre a continuidade de cultiva-lo



Produto  Faturamento
2            Café    2581452.0
0         Algodão    1499494.0
5            Soja     167706.0
6           Trigo     116768.0
1           Arroz     115323.0
3  Cana-de-açúcar      90150.0
4           Milho      73801.0

In [177]:
print(faturamento_total.sum())
ft = faturamento_total.sum()
ft
#Faturamento total obtido das vendas dos produtos

Produto        CaféAlgodãoSojaTrigoArrozCana-de-açúcarMilho
Faturamento                                       4644694.0
dtype: object


Produto        CaféAlgodãoSojaTrigoArrozCana-de-açúcarMilho
Faturamento                                       4644694.0
dtype: object

## Itens não vendidos(Em estoque)

In [178]:
# Calculando a perda de cada linha
df["Não_vendidos"] = df['Estoque'] * df["Preco_kg"]

# Agrupando pelo produto
Perdas = df.groupby('Produto')['Não_vendidos'].sum().round(2).reset_index().sort_values(by="Não_vendidos", ascending=False)

print(Perdas)



          Produto  Não_vendidos
2            Café    4960333.42
0         Algodão    2682745.52
5            Soja     323352.74
6           Trigo     216905.00
1           Arroz     212977.44
3  Cana-de-açúcar     178818.56
4           Milho     138740.65


In [179]:
print(Perdas.sum())
p = Perdas.sum()
p

Produto         CaféAlgodãoSojaTrigoArrozCana-de-açúcarMilho
Não_vendidos                                      8713873.33
dtype: object


Produto         CaféAlgodãoSojaTrigoArrozCana-de-açúcarMilho
Não_vendidos                                      8713873.33
dtype: object

In [180]:
print(faturamento_total.sum(),Perdas.sum())

Produto        CaféAlgodãoSojaTrigoArrozCana-de-açúcarMilho
Faturamento                                       4644694.0
dtype: object Produto         CaféAlgodãoSojaTrigoArrozCana-de-açúcarMilho
Não_vendidos                                      8713873.33
dtype: object


In [181]:
Indice = 4644694.0 / 8713873.33
Indice
# Através desse calculo sobre o faturamneto total e perdas descobrimos que estamos estocamndo mais doque vendendendo 

0.5330228962600722

### Cidade com maiores Faturamentos e vendederoes com maiores Faturamentos

In [182]:
City_F = df.groupby('cidade')['Faturamento'].sum().reset_index().round().sort_values(by = 'Faturamento', ascending=False)
City_F

cidade  Faturamento
9       São Paulo     555279.0
4          Manaus     494631.0
7  Rio de Janeiro     481187.0
6          Recife     481039.0
3       Fortaleza     477431.0
1        Brasília     455123.0
8        Salvador     431106.0
2        Curitiba     428082.0
0  Belo Horizonte     426561.0
5    Porto Alegre     414254.0

In [203]:
sales_F = df.groupby('Vendedor')['Faturamento'].sum().reset_index().round().sort_values(by = 'Faturamento', ascending=False)
sales_F

Vendedor  Faturamento
6      Patrícia     684681.0
3      Fernanda     612031.0
7         Pedro     606358.0
0           Ana     530639.0
1        Carlos     523892.0
5         Maria     500050.0
8       Ricardo     499178.0
4          João     387170.0
2  Desconhecido     300693.0

### Cidade com maiores Vendas e vendederoes com maiores Vendas

In [184]:
City_Maior = df.groupby('cidade')['Vendidos'].sum().reset_index().round().sort_values(by = 'Vendidos', ascending=False)
City_Maior
#Constatamos que o Rio de Janeiro e Fortaleza foram os estados que mais venderam

cidade  Vendidos
7  Rio de Janeiro     59943
3       Fortaleza     56701
9       São Paulo     56548
6          Recife     52194
2        Curitiba     51371
8        Salvador     50931
4          Manaus     49572
1        Brasília     48192
5    Porto Alegre     47138
0  Belo Horizonte     45422

In [185]:
sales_Maior = df.groupby('Vendedor')['Vendidos'].sum().reset_index().round().sort_values(by = 'Vendidos', ascending=False)
sales_Maior
# Constatamos que Patrícia e Ricardo foram os vendedores que mais venderam

Vendedor  Vendidos
6      Patrícia     63862
8       Ricardo     63138
3      Fernanda     62766
7         Pedro     61875
1        Carlos     61138
0           Ana     60070
5         Maria     57919
4          João     55819
2  Desconhecido     31425

### Data com maiores vendas

In [186]:
df['Data_de_vendas'] = pd.to_datetime(df['Data_de_vendas'], errors='coerce')
# Transforma no formato de data


In [187]:
date_Maior = df.groupby('Data_de_vendas')['Vendidos'].sum().reset_index().round().sort_values(by = 'Vendidos', ascending=False)
date_Maior
# Percebemos que o dia 14 do mes 10 de 2025 foi o dia que mias obtivemos venda e o restante dos dias as vendas foram muito espalhadas

Data_de_vendas  Vendidos
646      2025-10-14      1372
552      2025-07-12       998
1596     2028-05-27       996
1337     2027-09-09       992
1943     2029-05-09       989
...             ...       ...
137      2024-05-18         1
374      2025-01-13         1
213      2024-08-03         1
419      2025-02-28         1
872      2026-05-30         1

[1988 rows x 2 columns]

### Por Ano

In [188]:
faturamento_ano = df.groupby(df['Data_de_vendas'].dt.year)['Vendidos'].sum().reset_index().sort_values(by="Vendidos", ascending=False)
print(faturamento_ano)
# Percebemos que o ano que mais lucramos foi o entre 2028 e 2024 e que de 24 e 25 foi aumentando até 2028, e que houve algo que fez as vendas cairem quase pela metade, observar oque houve para maximizar as vendas 



   Data_de_vendas  Vendidos
4            2028     97384
2            2026     95509
3            2027     94240
0            2024     93501
1            2025     93208
5            2029     44170


In [189]:
faturamento_Mês = df.groupby(df['Data_de_vendas'].dt.month)['Vendidos'].sum().reset_index().sort_values(by="Vendidos", ascending=False)
print(faturamento_Mês)
# Percebemos que o mês 2 e 5 venderam bem e que foi se mantendo mais reduzindo as vendas, demonstrando uma estabilidade de certa forma nas vendas

    Data_de_vendas  Vendidos
1                2     48655
4                5     48410
0                1     46809
3                4     46310
11              12     43011
9               10     42923
7                8     42843
2                3     41660
5                6     41127
10              11     38918
8                9     38850
6                7     38496


In [190]:
faturamento_dia = df.groupby(df['Data_de_vendas'].dt.day)['Vendidos'].sum().reset_index().sort_values(by="Vendidos", ascending=False)
print(faturamento_dia)
# Nota-se que o dia 5 em específico tem uma grande procura , quase 2mil vendas na frende da segunda data, devemos observar o dia 5 para saber por qual motivo ele vende mais

    Data_de_vendas  Vendidos
4                5     20418
8                9     18914
1                2     18556
0                1     18344
11              12     18157
16              17     18126
6                7     17892
27              28     17580
23              24     17527
3                4     17446
19              20     17121
10              11     17091
13              14     17034
17              18     17014
28              29     16930
12              13     16911
24              25     16814
5                6     16796
20              21     16707
2                3     16521
22              23     16413
18              19     16385
9               10     16366
15              16     16079
21              22     15523
14              15     15514
7                8     15209
25              26     14805
29              30     14766
26              27     13892
30              31     11161


In [191]:
import sys
print(sys.executable)


c:\Users\luisf\AppData\Local\Programs\Python\Python313\python.exe


In [ ]:
df_cidade = df.groupby("cidade", as_index=False).agg({
    "Faturamento": "sum",   
    "Vendidos": "sum",      
    "latitude": "first",    
    "longitude": "first"    
})

import plotly.express as px

fig = px.scatter_map(
    df_cidade,
    lat="latitude",
    lon="longitude",
    size="Faturamento",      
    color="Faturamento",     
    hover_name="cidade",
    hover_data=["Vendidos", "Faturamento"],  # mostra vendas totais e faturamento
    size_max=30,
    zoom=4
)

fig.update_layout(map_style="carto-darkmatter")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()


# Representação gráfica com plotly.express das cidade com maior faturamento



In [ ]:
df_vendedor = df.groupby("Vendedor", as_index=False).agg({
    "Faturamento": "sum",    
    "Vendidos": "sum",      
    "latitude": "first",    
    "longitude": "first"  
})

import plotly.express as px

fig = px.scatter_map(
    df_vendedor,
    lat="latitude",
    lon="longitude",
    size="Faturamento",     
    color="Faturamento",      
    hover_name="Vendedor",  
    hover_data=["Vendidos", "Faturamento"],  
    size_max=30,
    zoom=4
)

fig.update_layout(map_style="carto-darkmatter")
fig.update_layout(margin={"r":0,"t":0,"l":0,"b":0})
fig.show()

# Representação gráfica com plotly.express das Vendedores com maior faturamento


### Outras estatistícas básicas

In [193]:
df['Vendidos'].mean()
# Temoos que a média de vendas ficou por 259 itens diariamente

np.float64(259.006)

In [194]:

df['Estoque'].mean()
# Temoos que a média de itens não vendidos ficou por 488 itens diariamente

np.float64(487.692)

In [195]:
df['Produto'].mode()
# Temos que o produto que mais vendeu foi

0    Algodão
Name: Produto, dtype: object

In [196]:
df['Categoria'].unique()

array(['Cereal', 'Grão', 'Bebida', 'Fibra'], dtype=object)

In [197]:
df.drop(columns=["Categoria"], inplace=True)
df
# Removemos a coluna "categoria pois ela estava irrelevante para a análise e trazia dados inconsistentes demias para tratar"

Produto  Estoque  Vendidos Data_de_vendas      Vendedor  \
0               Café      466       277     2024-01-01  Desconhecido   
1               Soja      191        37     2024-01-02         Pedro   
2     Cana-de-açúcar      241       241     2024-01-03      Fernanda   
3              Arroz      222       283     2024-01-04        Carlos   
4              Arroz      370       432     2024-01-05      Patrícia   
...              ...      ...       ...            ...           ...   
1995           Arroz      493       336     2029-06-18      Patrícia   
1996  Cana-de-açúcar      457       385     2029-06-19           Ana   
1997         Algodão      316       276     2029-06-20         Maria   
1998           Trigo       65       319     2029-06-21          João   
1999           Milho      982       403     2029-06-22       Ricardo   

              cidade   latitude  longitude  Preco_kg  Faturamento  \
0          São Paulo -23.550520 -46.633308     38.57     10683.89   
1     Rio de Janeiro -22.906847 -43.172896      2.33        86.21   
2           Brasília -15.794229 -47.882166      1.28       308.48   
3     Belo Horizonte -19.924502 -43.935237      1.56       441.48   
4          Fortaleza  -3.717220 -38.543400      1.56       673.92   
...              ...        ...        ...       ...          ...   
1995       São Paulo -23.550520 -46.633308      1.56       524.16   
1996    Porto Alegre -30.034647 -51.217658      1.28       492.80   
1997          Recife  -8.047560 -34.877000     17.68      4879.68   
1998        Salvador -12.971399 -38.501305      1.42       452.98   
1999          Recife  -8.047560 -34.877000      1.09       439.27   

      Não_vendidos  
0         17973.62  
1           445.03  
2           308.48  
3           346.32  
4           577.20  
...            ...  
1995        769.08  
1996        584.96  
1997       5586.88  
1998         92.30  
1999       1070.38  

[2000 rows x 11 columns]

## Estatistica básica (Desvio padrão, Vareiância e coeficiente de variação)
- Variância: É a média dos quadrados entre cada valor e a média. Informa quão distantes os valores estão da média 
- Desvio Padrão: É a raiz quadrada da variância Expressa a dispersão em termos da mesma unidade de dados
- Coeficiente de variação:É o desvio padrão dividido pela média, expresso em porcentagem. Indica a variabilidade relativa dos dados em relação a média

In [198]:
df['Vendidos'].var()
# Variância de vendidos, está muito alto, ou seja os dados estão bem diferentes entre si(inconsatantes)

np.float64(30134.226077038522)

In [199]:
df['Vendidos'].std()  # Tem uma diferença de mais ou menos 173 vendas entre si seja pra mias seja pra menos

np.float64(173.59212561933367)

In [200]:
df['Vendidos'].std() / df['Vendidos'].mean() * 100
#A dispersão dos dados é considerável, mas não exagerada. Em média, cada valor de vendas difere da média uma boa parte da média (mais da metade).
#Isso confirma que os dados não são super consistentes, mas também não são totalmente caóticos.


np.float64(67.02243408235087)

## Medidas de Forma (distribuição dos dados)
- Assimetria: Indica o grau e direção da distorção da distribuição em relação à média. Se positivo, indica a cauda direita da distribuição mais longa, já o oposto a cauda esquerda mais longa(Quando for positivo valores maiores mais distantes, quando for negativo mais perto eles estão).
- Curtose: mede o pico ou "pontuação" da distribuição. Uma Curtose alta indica uma distribuição mais concentrada 

In [201]:
df['Vendidos'].skew() 
# Essa assimetria perto de 1 indica altos graus dispersão, ou seja os dados estão m,uito espalhados

np.float64(0.9595618887978993)

In [202]:
df['Vendidos'].kurtosis()
# Como a curtose estrá alta isso é bom, indica que estamos mais próximos da média de vendas e que mantemos uma certa "estabilidade" nas vendas 

np.float64(1.84885137858533)

# 📊 Análise de Dados de Vendas Agrícolas

## Conclusão

Este projeto teve como objetivo analisar os dados de vendas de produtos agrícolas, utilizando Python e Pandas, com foco em **insights financeiros e geográficos**.  

### 1️⃣ Estatísticas de vendas

- **Faturamento total por produto:**  
  Foi possível calcular o faturamento total de cada produto a partir da multiplicação de `Vendidos` pelo `Preco_kg`.  
  Produtos como **Café e Algodão** se destacaram com faturamentos maiores.

- **Perdas com produtos não vendidos:**  
  Multiplicando `Estoque` pelo `Preco_kg`, identificamos os produtos com maiores perdas. Por exemplo, Café apresentou as maiores perdas devido ao alto preço e volume em estoque.

- **Variabilidade das vendas:**  
  - **Variância de `Vendidos`:** 30.134 → indica que os dados apresentam dispersão significativa.  
  - **Desvio Padrão:** 173,59 → em média, cada registro de vendas difere ±174 unidades da média.  
  - **Coeficiente de Variação:** 67% → mostra que existe variabilidade considerável nas vendas em relação à média, indicando que nem todos os produtos vendem de forma uniforme.

### 2️⃣ Análise temporal

- As vendas foram agrupadas por **dia, mês e ano**, permitindo observar sazonalidade e tendências ao longo do tempo.

### 3️⃣ Análise geográfica

- Utilizando **Plotly Express**, foi possível criar mapas interativos mostrando:
  - **Faturamento total por cidade:** cada cidade aparece uma única vez, com pontos cujo tamanho e cor refletem o faturamento total.  
  - **Faturamento por vendedor:** cada vendedor é representado por um ponto, mostrando o faturamento total que gerou e os produtos vendidos.  

Esses mapas permitem identificar **as regiões com maior desempenho de vendas** e quais vendedores estão contribuindo mais para o faturamento.

### 4️⃣ Insights gerais

- Os dados apresentam **vendas inconsistentes entre produtos,cidades e vendedores**, com algumas regiões e produtos dominando o faturamento.  
- É possível **priorizar produtos e regiões** com maior retorno financeiro, além de identificar **oportunidades de reduzir perdas** em estoque.  
- A visualização geográfica facilita **decisões estratégicas** para logística e planejamento de vendas.

---

📌 **Resumo:** Este projeto demonstra a importância de combinar **estatísticas descritivas** com **visualizações interativas**, oferecendo uma visão clara do desempenho de produtos, vendedores e regiões no negócio agrícola, concluindo-se que a análise da forma como estamos estocando alimentos ser repensada.
